libs/agno/agno/workflow/v2/pipeline.py

```diff
@@ -85,24 +79,107 @@ def execute(self, inputs: Dict[str, Any], context: Dict[str, Any] = None) -> Ite
             # Update previous_outputs for next task
             self._update_previous_outputs(previous_outputs, task, task_output, i)
 
-            # Task completed event
-            yield TaskCompletedEvent(
-                run_id=context.get("run_id", ""),
-                content=task_output.content,
-                workflow_name=context.get("workflow_name") if context else None,
-                pipeline_name=self.name,
-                task_name=task.name,
-                task_index=i,
-                workflow_id=context.get("workflow_id") if context else None,
-                session_id=context.get("session_id") if context else None,
-                images=task_output.images,
-                videos=task_output.videos,
-                audio=task_output.audio,
-                messages=getattr(task_output.response, "messages", None) if task_output.response else None,
-                metrics=getattr(task_output.response, "metrics", None) if task_output.response else None,
-                task_responses=[task_output],
-            )
+        # Create final output data
+        final_output = {
+            "pipeline_name": self.name,
+            "pipeline_id": self.pipeline_id,
+            "status": "completed",
+            "total_tasks": len(self.tasks),
+            "task_summary": [
+                {
+                    "task_name": task.name,
+                    "task_id": task.task_id,
+                    "description": task.description,
+                    "executor_type": task.executor_type,
+                    "executor_name": task.executor_name,
+                }
+                for task in self.tasks
+            ],
+        }
+
+        return WorkflowRunResponse(
+            event=WorkflowRunEvent.workflow_completed,
+            content=f"Pipeline {self.name} completed successfully",
+            workflow_id=context.get("workflow_id") if context else None,
+            workflow_name=context.get("workflow_name") if context else None,
+            pipeline_name=self.name,
+            run_id=context.get("run_id", ""),
+            session_id=context.get("session_id") if context else None,
+            task_responses=collected_task_outputs,
+            extra_data=final_output,
+        )
+
+    def _execute_stream(
+        self, inputs: Dict[str, Any], context: Dict[str, Any] = None, stream_intermediate_steps: bool = False
```

Comment:
We should create a WorkflowRunResponse when the workflow run starts. Then you pass that down through pipeline (instead of `context`) and you reference the run_id, name, session_id etc from there. See how we do it in agent/team.

On Task I can see the need for the context, it is easier to work with for a user, but not on sequence